src/limiter.rs

```diff
@@ -294,7 +301,7 @@ impl<C: Clock> Limiter<C> {
 
     /// Consumes several bytes from the speed limiter, returns the duration
     /// needed to sleep to maintain the speed limit.
-    fn consume_duration(&self, byte_size: usize) -> Duration {
+    pub fn consume_duration(&self, byte_size: usize) -> Duration {
```

Comment:
why making this public

Reply:
I need a duration instead of a future to check some states within the time span periodically.